research/picai/preprocessing_transforms.py

```diff
@@ -157,13 +157,14 @@ def crop_or_pad(
     # input conversion and verification
     shape, size = input_verification_crop_or_pad(image, size, physical_size)
 
-    # set identity operations for cropping and padding
-    rank = len(size)
-    padding = [[0, 0]] * rank
-    slicer = [slice(None)] * rank
+    # Since the subarrays are being set in the below for loop
+    # We have to ensure that they are seperate lists
+    # and not the same reference (ie [[0, 0]] * rank)
+    padding = [[0, 0] for _ in range(len(size))]
```

Comment:
Ah interesting. I didn't realize this tied them through reference like that. It doesn't work that way for things like `ints`. Good to know for the future.

Reply:
Yeah I did not realize it either, I think it is only for mutable types